### ISSUE TITLE
Line plot not autoscaled

### ISSUE THREAD

**Issue by bfkeats** [pandas-dev/pandas#17684]
[dat.zip](https://github.com/pandas-dev/pandas/files/1333773/dat.zip)

edit: Attached the data to reproduce

#### 

```python
import pandas as pd
import matplotlib.pyplot as plt
DF = pd.read_csv('dat.csv')
# Not autoscaled
DF.plot('x','y')
plt.show()
# Autoscaled
ax1 = DF.plot('x','y')
ax1.autoscale(True)
plt.show()
# Autoscaled
plt.plot(DF['x'],DF['y'])
plt.show()
```
#### Problem description

When I create a line plot with this particular dataset, the x-axis does not scale correctly.

#### Expected Output
The autoscaled plot.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.1
pytest: 3.0.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
xarray: None
IPython: 5.3.0
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.2.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.6.0
html5lib: 0.999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None

</details>

**Comment by TomAugspurger** [comment]
Can you provide a reproducible example?

**Comment by bfkeats** [comment]
Sorry, forgot to attached the csv. Added to original post.

**Comment by TomAugspurger** [comment]
Typically it's easier for us to diagnose what's going on if you avoid extra
files. Can you create a small DataFrame from dictionaries / lists that
demonstrates the issue?

**Comment by bfkeats** [comment]
When I create a small dataframe, it autoscales fine. It has something to do with this particular data.

**Comment by sinhrks** [comment]
Then, is it possible for you to upload 'dat.csv' to somewhere? Otherwise, it is difficult for us to investigate it.

**Comment by bfkeats** [comment]
I attached it to the original post.

**Comment by nmartensen** [comment]
This is fixed in v0.21.0.

Example code:
```
angles = np.arange(1, 11)/5. * np.pi
df = pd.DataFrame({'c': np.cos(angles), 's': np.sin(angles)})
df.plot(x='c', y='s')
```
old output:
![pandas_old](https://user-images.githubusercontent.com/16715246/32579351-d75275a4-c4e0-11e7-9987-ada4162d2701.PNG)

new output:
![pandas_new](https://user-images.githubusercontent.com/16715246/32579365-e46e460a-c4e0-11e7-8d65-5cd6c59d3cd1.PNG)

**Comment by TomAugspurger** [comment]
Thanks, closed by https://github.com/pandas-dev/pandas/pull/16600.
